src/ephy-shell.c

```diff
@@ -554,12 +555,18 @@ static void
 ephy_shell_init (EphyShell *shell)
 {
   EphyShell **ptr = &ephy_shell;
+  EphySyncService *service;
 
   /* globally accessible singleton */
   g_assert (ephy_shell == NULL);
   ephy_shell = shell;
   g_object_add_weak_pointer (G_OBJECT (ephy_shell),
                              (gpointer *)ptr);
+
+  /* Start the periodical sync now. */
+  service = ephy_sync_service_new ();
+  ephy_sync_service_start_periodical_sync (service, TRUE);
```

Comment:
@mcatanzaro I'm also calling `ephy_sync_service_start_periodical_sync` from `prefs-dialog.c` to start the sync once the user has logged in.